urwid/container.py

```diff
@@ -103,6 +103,18 @@ def get_focus_widgets(self):
                 return out
             out.append(w)
 
+    def focus_first_selectable(self):
+        """
+        Set the focus of the container to the selectable child widget
+        that comes first in its hierarchy.
+        """
+        for j in range(0, len(self.contents)):
```

Comment:
positions aren't always integers, so it would be better to start from the current focus and get the next child until you find a widget that is selectable. You'll also want to give up after some fixed number because containers may also have an unbounded number of widgets (and might be reading widgets off network or disk when they are iterated over)